xwiki-platform-core/xwiki-platform-bridge/src/main/java/org/xwiki/bridge/event/UserAuthenticatedEvent.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.bridge.event;
+
+import org.xwiki.model.reference.DocumentReference;
+
+/**
+ * An event triggered when a user is authenticated.
+ * <p>
+ * The event also send the following parameters:
+ * </p>
+ * <ul>
+ * <li>source: the current {com.xpn.xwiki.user.api.XWikiUser} instance which has been authenticated.</li>
+ * <li>data: the current {com.xpn.xwiki.XWikiContext} instance</li>
```

Comment:
Don't. There is no point in passing the XWikiContext. Those which does it are very old events and keep doing that for retro compatibility reasons.

Reply:
What is a good example of recent Event management ?

Reply:
Depends on which aspect but regarding the parameters the idea is that we should pass only things which bring value. The reason why old events are passing the XWikiContext is because it was the only way for the listener to access it basically while it's easy now using the Provider<XWikiContext>.

In general if you are not sure if something worth being sent with the event don't send it. It's very easy to add something as a parameter later (as long as the event is not already associated with two parameters), but it's very hard to remove it since that would be an API breakage.

Reply:
Okay, I fully understand the problem now. I'll take care to avoid such a thing next times.